### ISSUE TITLE
More detailed error reporting on import fails

### ISSUE THREAD

**Issue by georgehrke** [nextcloud/calendar#2065]
**Is your feature request related to a problem? Please describe.**

There are plenty of broken ics files out there. We already try our best to fix them with [the repair-steps](https://github.com/georgehrke/calendar-js/tree/c05059cf72ad3d632797ddeb0939e7b615d41252/src/parsers/repairsteps/icalendar).
Nonetheless, we can't fix everything.

**Describe the solution you'd like**
Parse the response message when import fails and check for a Sabre error code:
- `UnsupportedMediaType` or `ParseException` for syntax errors
- `NoInstancesException` for error like this one
- Are there more relevant?

We can use the responses for more detailed error messages in the console.
If we end up with a `NoInstancesException`, we might just count it as a successful import, because it contains no events to import anyway.

**Comment by georgehrke** [comment]
We could also check for empty recurrence-sets in the client: https://github.com/georgehrke/calendar-js/blob/c05059cf72ad3d632797ddeb0939e7b615d41252/src/recurrence/recurrenceManager.js#L566

**Comment by mrbenjoi** [comment]
A more detailed error message might also be important for the user.

I just wanted to import some calendars I previously exported from Google, and everyhting works fine except for one event. The error message says, that 154 of 155 events were imported.

At this point I would really like to know WHICH event it was, that was not imported.

Is it possible to also make the error message in the UI more detailed. Maybe it is sensible to show a dialog with a text area containing all events that were not imported correctly. The user can then decide whether these events are important or not and add them manually (as long as they are few of course).

**Comment by Matthias84** [comment]
Yes, also re-importing creates non-informative notifications. Are the items skipped because they are dublicates, or because of format errors? :smiley:

**Comment by r2evans** [comment]
A related issue in server https://github.com/nextcloud/server/issues/30096 was closed last year, though it is still happening. Can a dev comment please?
